### PR TITLE
Honor HAL_LOGGING_ENABLED

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

File: ArduPlane/Log.cpp [hunk 1]
```diff
@@ -470,8 +471,8 @@ const struct LogStructure Plane::log_structure[] = {
       "CMDA", "QHBBBBffffiifB",    "TimeUS,CId,TSys,TCmp,cur,cont,Prm1,Prm2,Prm3,Prm4,Lat,Lng,Alt,F", "s---------DUm-", "F---------GGB-" }, 
     { LOG_CMDH_MSG, sizeof(log_CMDI),     
       "CMDH", "QHBBBBffffiifB",    "TimeUS,CId,TSys,TCmp,cur,cont,Prm1,Prm2,Prm3,Prm4,Lat,Lng,Alt,F", "s---------DUm-", "F---------GGB-" }, 
-
 };
+#endif
```
Comment (khancyr) [inline ArduPlane/Log.cpp:475]: this isn't good it breaks the #else at L535

**Review by khancyr — CHANGES_REQUESTED** [review]
Good idea but need fixing

File: ArduPlane/Log.cpp (hunk 1 above)
Comment (khancyr) [reply ArduPlane/Log.cpp:475]: #else
const struct LogStructure Plane::log_structure[]{};
#endif

Comment (peterbarker) [reply ArduPlane/Log.cpp]: Fixed - moved to using `HAL_LOGGING_ENABLED` everywhere rather than a mix of `HAL_LOGGING_ENABLED` and `LOGGING_ENABLED`

**Comment by tridge** [comment]
this would be nice
AP_LOGGER_WRITE()
matching::
AP::logger().write()

**Comment by rmackay9** [comment]
I'll just bang on a bit more than we should always have a description in the PR

**Review by khancyr — APPROVED** [review]
LGTM

**Comment by khancyr** [comment]
@peterbarker anything blocking this one ?

**Comment by peterbarker** [comment]
On Tue, 10 Aug 2021, Pierre Kancir wrote:

> @peterbarker anything blocking this one ?

We should probably close this one.

We'll be going with something like we do for gcs().send_text 
-> GCS_SEND_TEXT instead to avoid wasting flash even on the dummy class.

So LOGGER_WRITE(...) and a few friends.

**Pull request closed without merging**